### ISSUE TITLE
Tabs Valid Parent

### ISSUE THREAD

**Issue by Alek99** [reflex-dev/reflex#2173]
**Describe the bug**

When using tabs_list and tab_panel outside of tabs we should through an error message at compile time

✅ Valid
```python
rx.tabs(
            rx.tab_list(
                rx.tab("Learn"),
                rx.tab("Concepts"),
            ),
            rx.tab_panels(
                rx.tab_panel(
                    rx.heading(
                        rx.span("[ ", color="#DACEEE"),
                        "Learn",
                        rx.span(" ]", color="#DACEEE"),
                        style=heading_style3,
                    ),
                    rx.accordion(
                        *[
                            sidebar_item_comp(
                                item=item,
                                index=[-1],
                                url=url,
                                first=True,
                            )
                            for item in learn
                        ],
                        allow_multiple=True,
                        default_index=learn_index,
                    ),
                    rx.heading(
                        rx.span("[ ", color="#DACEEE"),
                        "Concepts",
                        rx.span(" ]", color="#DACEEE"),
                        style=heading_style3,
                        margin_top="1em",
                    ),
                    rx.accordion(
                *[
                    sidebar_item_comp(
                        item=item,
                        index=[-1],
                        url=url,
                        first=True,
                    )
                    for item in concepts
                ],
                allow_multiple=True,
                default_index=concepts_index,
            ),
                ),
                rx.tab_panel(   
                    rx.heading(
                        rx.span("[ ", color="#DACEEE"),
                        "Reference",
                        rx.span(" ]", color="#DACEEE"),
                        style=heading_style3,
                        margin_top="1em",
                    ),
                    rx.accordion(
                *[
                    sidebar_item_comp(item=item, url=url, first=True, index=reference_index)
                    for item in reference
                ],
                allow_multiple=True,
                default_index=reference_index,
            ),
                )
            )
        ),
```

**To Reproduce**
 - Code/Link to Repo:

Use tab_list and tab_panel outside of a tabs component. 

**Expected behavior**

**Screenshots**

Currently, I get this error in the browser.

![Screenshot 2023-11-14 at 5 39 21 PM](https://github.com/reflex-dev/reflex/assets/38776361/588b471f-11e5-47c7-a4b7-8f49d3c359bc)

**Specifics (please complete the following information):**
 - Python Version:
 - Reflex Version:
 - OS: 
 - Browser (Optional):
 
**Additional context**

**Comment by kisharnath** [comment]
Can I work on this?

**Comment by Mohit-majumdar** [comment]
can we close this issue? It was solved in #2463
